gobblin-runtime/src/main/java/org/apache/gobblin/runtime/api/InstrumentedSpecStore.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.runtime.api;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.concurrent.TimeUnit;
+
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import com.google.common.base.Optional;
+import com.typesafe.config.Config;
+
+import org.apache.gobblin.configuration.State;
+import org.apache.gobblin.instrumented.Instrumented;
+import org.apache.gobblin.metrics.GobblinMetrics;
+import org.apache.gobblin.metrics.MetricContext;
+import org.apache.gobblin.metrics.ServiceMetricNames;
+import org.apache.gobblin.util.ConfigUtils;
+
+
+/**
+ * Instrumented version of {@link SpecStore} automatically capturing certain metrics.
+ * Subclasses should implement addSpecImpl instead of addSpec and so on.
+ */
+public abstract class InstrumentedSpecStore implements SpecStore {
+  private Optional<Timer> getTimer;
+  private Optional<Timer> existsTimer;
+  private Optional<Timer> deleteTimer;
+  private Optional<Timer> addTimer;
+  private Optional<Timer> updateTimer;
+  private Optional<Timer> getAllTimer;
+  private Optional<Timer> getURIsTimer;
+  private MetricContext metricContext;
+  private final boolean instrumentationEnabled;
+
+  public InstrumentedSpecStore(Config config, SpecSerDe specSerDe) {
+    this.instrumentationEnabled = GobblinMetrics.isEnabled(new State(ConfigUtils.configToProperties(config)));
+    this.metricContext = Instrumented.getMetricContext(new State(), getClass());
+    if (instrumentationEnabled) {
+      this.getTimer = Optional.of(this.metricContext.timer(getClass() + "-GET"));
+      this.existsTimer = Optional.of(this.metricContext.timer(
+          MetricRegistry.name(ServiceMetricNames.GOBBLIN_SERVICE_PREFIX,getClass().getSimpleName(), "-EXISTS")));
```

Comment:
Can we separate these lines to a `createTimer(String suffix)` method to remove repeated code?